app/server/utils/ussd/token_processor.py

```diff
@@ -73,61 +74,53 @@ def get_default_limit(user: User, token: Token, transfer_account: TransferAccoun
             recipient_user=user,
             token=token,
-            amount=0)
-
-        limits = example_transfer.get_transfer_limits()
+            amount=0
+        ) as dummy_transfer:
 
-        try:
-            # Sometimes some weird db behaviour cause the above transfer to be persisted.
-            # This is here to make sure it definitely doesn't hang around
-            db.session.delete(example_transfer)
-        except Exception as e:
-            pass
+            limits = dummy_transfer.get_transfer_limits()
 
-        if len(limits) == 0:
-            return None
-        else:
-            # should only ever be one ge limit
-            ge_limit = [limit for limit in limits if 'GE Liquid Token' in limit.name]
-
-            lowest_limit = None
-            lowest_amount_avail = float('inf')
-
-            for limit in limits:
-                if 'GE Liquid Token' not in limit.name:
-                    transaction_volume = limit.apply_all_filters(
-                        example_transfer,
-                        db.session.query(func.sum(CreditTransfer.transfer_amount).label('total'))
-                    ).execution_options(show_all=True).first().total
-
-                    amount_avail = limit.total_amount - (transaction_volume or 0)
-                    fraction = ge_limit[0].transfer_balance_fraction or 0
-                    if amount_avail < (fraction * transfer_account.balance)\
-                            and amount_avail < lowest_amount_avail:
-                        lowest_limit = limit
-                        lowest_amount_avail = amount_avail
-
-            if lowest_limit:
-                return lowest_limit
+            if len(limits) == 0:
+                return None
             else:
-                return ge_limit[0]
+                # should only ever be one ge limit
+                ge_limit = [limit for limit in limits if 'GE Liquid Token' in limit.name]
+
+                lowest_limit = None
+                lowest_amount_avail = float('inf')
+
+                for limit in limits:
+                    if 'GE Liquid Token' not in limit.name:
+                        transaction_volume = limit.apply_all_filters(
+                            dummy_transfer,
+                            db.session.query(func.sum(CreditTransfer.transfer_amount).label('total'))
+                        ).execution_options(show_all=True).first().total
+
+                        amount_avail = limit.total_amount - (transaction_volume or 0)
+                        fraction = ge_limit[0].transfer_balance_fraction or 0
+                        if amount_avail < (fraction * transfer_account.balance)\
+                                and amount_avail < lowest_amount_avail:
+                            lowest_limit = limit
+                            lowest_amount_avail = amount_avail
+
+                if lowest_limit:
+                    return lowest_limit
+                else:
+                    return ge_limit[0]
```

Comment:
what's happened here @enjeyw ?

Reply:
nothing. That's my point above - the function is identical, the wrapping has just changed

Reply:
cool